cranelift/codegen/src/ir/exception_table.rs

```diff
@@ -127,38 +191,16 @@ impl ExceptionTableData {
 
     /// Get the targets for exceptional return cases, together with
     /// their tags.
-    pub fn catches(&self) -> impl Iterator<Item = (Option<ExceptionTag>, &BlockCall)> + '_ {
-        self.tags
-            .iter()
-            .map(|tag| tag.expand())
-            // Skips the last entry of `targets` (the normal return)
-            // because `tags` is one element shorter.
-            .zip(self.targets.iter())
-    }
-
-    /// Get the targets for exceptional return cases, together with
-    /// their tags.
-    pub fn catches_mut(
-        &mut self,
-    ) -> impl Iterator<Item = (Option<ExceptionTag>, &mut BlockCall)> + '_ {
-        self.tags
-            .iter()
-            .map(|tag| tag.expand())
-            // Skips the last entry of `targets` (the normal return)
-            // because `tags` is one element shorter.
-            .zip(self.targets.iter_mut())
-    }
-
-    /// The number of catch edges in this exception table.
-    pub fn len_catches(&self) -> usize {
-        self.tags.len()
-    }
-
-    /// Get the `index`th catch edge from this table.
-    pub fn get_catch(&self, index: usize) -> Option<(Option<ExceptionTag>, &BlockCall)> {
-        let tag = self.tags.get(index)?.expand();
-        let target = &self.targets[index];
-        Some((tag, target))
+    pub fn catches(&self) -> impl Iterator<Item = ExceptionTableItem> + '_ {
+        self.items.iter().map(|item| match item {
+            InternalExceptionTableItem::Tag(tag, target_idx) => {
+                ExceptionTableItem::Tag(*tag, self.targets[usize::try_from(*target_idx).unwrap()])
+            }
+            InternalExceptionTableItem::Default(target_idx) => {
+                ExceptionTableItem::Default(self.targets[usize::try_from(*target_idx).unwrap()])
+            }
+            InternalExceptionTableItem::Context(ctx) => ExceptionTableItem::Context(*ctx),
+        })
```

Comment:
This method would previously give only exceptional edges, but now gives all kinds of table items. That seems fine if callers were not relying on that property, but we should definitely update the method documentation and probably the method name as well.